.github/workflows/remote-build-android-hybrid.yml

```diff
@@ -0,0 +1,30 @@
+name: Remote Build Android Hybrid
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [main]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest-xl
+    steps:
+      - name: Checkout
+        # v4
+        uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608
+        with:
+          submodules: true
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
This doesn't look right, because the `event` is not `pull_request`. I think you can just omit the ref key here and it will check out the ref that was pushed to.